docs/manual/upgrading-to-phpspec-3.rst

```diff
@@ -0,0 +1,47 @@
+Upgrading to PHPSpec 3
+======================
+
+PHPSpec 3.0 is still in development, but here is a guide to upgrading a test
+suite or an extension, based on BC-breaking changes made so far.
+
+Upgrading for Users
+-------------------
+
+So far, there are no changes that need to be made to get a typical test suite
+written for PHPSpec 2 to work with 3.0. We suggest trying the upgrade and
+running your tests to see what happens.
+
+If you are using 3rd party PHPSpec extensions, you may have to increase the
+version numbers for those as well.
+
+As PHP 5.5 and below are no longer supported language versions, you will need
+to upgrade to PHP 5.6 or 7.0+ to use PHPSpec 3.
+
+Upgrading for Extension Authors
+-------------------------------
+
+Several interfaces have been renamed in PHPSpec 3.0.  Here is a quick guide to
+changes you will need to make in your code.
+
+- *PhpSpec\Console\IO* is now *PhpSpec\Console\ConsoleIO*
```

Comment:
Have a look at the closed PRs, there's one by @shanethehat that does most of these changes. It might make sense to link to that PR, or make a separate docs page for this big list if it's too much to have inline in this page.